### PR TITLE
Collapses ComposableRecordStore into CommonAbstractStore

### PR DESCRIPTION
**Pull request by tinwelint**
because CommonAbstractStore already had knowledge about RecordFormat
at this point and so could implement the (previously) abstract methods
on its own. This results in shallower inheritance in store classes
and may make JITs job easier.

As a side effect MetaDataStore had to be changed to act as a real
record store in that it now has a proper MetaDataRecordFormat.
It still have the static record accessors, but that can be improved later.

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/impl/event/TransactionEventsIT.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 public class TransactionEventsIT
 {
     public final @Rule DatabaseRule db = new ImpermanentDatabaseRule();
-    public final @Rule RandomRule random = new RandomRule();
+    public final @Rule RandomRule random = new RandomRule().withSeed( 1457447436762L );
```
Comment (chrisvest) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/event/TransactionEventsIT.java]: Did you mean to fixate this seed?

**Comment by chrisvest** [comment]
> As a side effect MetaDataStore had to be changed to act as a real record store

It's for the best. Special snowflake stores are bug-magnets and a pain for our tooling.

File: community/kernel/src/test/java/org/neo4j/kernel/impl/event/TransactionEventsIT.java (hunk 1 above)
Comment (tinwelint) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/event/TransactionEventsIT.java]: Oops

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java [hunk 2]
```diff
@@ -735,7 +744,10 @@ private boolean isJustLegacyStoreTrailer( PageCursor cursor, int offset, byte[]
         }
     }
 
-    protected abstract int determineRecordSize();
+    protected int determineRecordSize()
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java:747]: Method can probably be private.

**Comment by lutovich** [comment]
LGTM. Can be merged when green.

**Pull request merged**